src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelperOptions.cs

```diff
@@ -23,21 +26,36 @@ public class HtmlHelperOptions
         /// <summary>
         /// Gets or sets the <see cref="string"/> that replaces periods in the ID attribute of an element.
         /// </summary>
-        public string IdAttributeDotReplacement { get; set; } = "_";
+        public string IdAttributeDotReplacement
+        {
+            get
+            {
+                return _idAttributeDotReplacement;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
```

Comment:
`throw`ing when we call `GetFullHtmlFieldName()` was too late. Much more common for users to change `HtmlHelperOptions` than for them to call that method directly.